src/meta/processors/admin/AdminClient.cpp

```diff
@@ -600,24 +600,43 @@ folly::Future<Status> AdminClient::getLeaderDist(HostLeaderMap* result) {
     return future;
 }
 
-folly::Future<Status> AdminClient::createSnapshot(GraphSpaceID spaceId,
-                                                  const std::string& name,
-                                                  const HostAddr& host) {
+folly::Future<StatusOr<std::string>> AdminClient::createSnapshot(GraphSpaceID spaceId,
+                                                                 const std::string& name,
+                                                                 const HostAddr& host) {
     if (injector_) {
         return injector_->createSnapshot();
     }
 
-    storage::cpp2::CreateCPRequest req;
-    req.set_space_id(spaceId);
-    req.set_name(name);
-
-    folly::Promise<Status> pro;
+    folly::Promise<StatusOr<std::string>> pro;
     auto f = pro.getFuture();
-    getResponse({Utils::getAdminAddrFromStoreAddr(host)}, 0, std::move(req),
-                [] (auto client, auto request) {
-                    return client->future_createCheckpoint(request);
-                },
-                0, std::move(pro), 3 /*The snapshot operation need to retry 3 times*/);
+
+    auto* evb = ioThreadPool_->getEventBase();
+    auto storageHost = Utils::getAdminAddrFromStoreAddr(host);
+    folly::via(evb, [evb, storageHost, pro = std::move(pro), spaceId, name, this]() mutable {
```

Comment:
Why don't use `getResponse`?

Reply:
This is because the current createSnapshot interface needs to return the corresponding snapshot path.

Reply:
`getResponse` could support it, there is a `respGen`.

Reply:
Neither function of getResponse satisfies the condition. The first function (4 arguments) requires that Request must contain a part id, and the second requires that the return value must be "Status".

Reply:
Got. thx